elastic/kibana
# [Security Solution][Lists] - Fix exception list with comments import bug

## Summary

Addresses https://github.com/elastic/kibana/issues/124742

#### Issue TLDR
Import of rules that reference exception items with comments fail. Failure message states that comments cannot include `created_at`, `created_by`, `id`.

#### Resolution TLDR
When importing rules, we do not trust these meta fields that are passed in, instead they are generated by the system. So on import, rules are set to show `created_by` as the person importing and `id`s are created anew.

The same pattern has been adopted here for exception item comments.

### Example import of comments
```js
 {
    comments: [
       // Example of an existing comment
      {
        comment: 'This is an exception to the rule',
        created_at: '2022-02-04T02:27:40.938Z',
        created_by: 'elastic',
        id: '845fc456-91ff-4530-bcc1-5b7ebd2f75b5',
      },
     // Example of a new comment importer may be wanting to add
      {
        comment: 'I decided to add a new comment',
      },
    ]
}
```

### After import comments look like:
```js
{
  [
      {
        comment: 'This is an exception to the rule',
        created_at: `${exceptionItemBody.comments[0].created_at}`,
        created_by: 'elastic',
        id: `${exceptionItemBody.comments[0].id}`,
      },
      {
        comment: 'I decided to add a new comment',
        created_at: `${exceptionItemBody.comments[2].created_at}`,
        created_by: 'elastic',
        id: `${exceptionItemBody.comments[2].id}`,
      },
    ]
}
```

### To test

- Create a rule
- Add an exception item with a comment 
- Export rule
- Import rule, selecting "overwrite" for rules and exceptions
- Should not see any error messages on import and exception item should still contain comment, though UI will now reflect that it was created by importer


### Checklist
